src/getProfile.js

```diff
@@ -0,0 +1,40 @@
+import { fetchProfiles } from './utils';
+
+const HOUR = 60 * 60
+
+export const getProfile = async (request, env, ctx) => {
+  const body = await request.json();
+  const relayUrls = body.relays.map(url => (new URL(url)).origin);
+  const pubkey = body.pubkey;
+
+  if (!relayUrls || !pubkey) {
+    return new Response('INVALID_REQUEST', { status: 422 });
+  }
+
+  const setCache = (key, data) => env.PROFILES.put(key, data, {
+    metadata: { createdAt: Date.now() },
+    // key expires after 2 days
+    expirationTtl: 2 * 24 * HOUR
+  })
+  const getCache = key => env.PROFILES.getWithMetadata(key)
+
+  let data;
+  const { value, metadata } = await getCache(pubkey)
+  if (value) {
+    data = JSON.parse(value)
+    // return if the data is NOT older than 3 hours
+    if (metadata.createdAt + 3 * HOUR * 1000 > Date.now()) {
```

Comment:
if we have some data in the cache we should always return that, but if it is older than x hours we should asynchronously update it in the background. 

I actually even think that we should always return the value from the cache (even if there is nothing) and if there is nothing in the cache or if it is too old we should trigger the refresh. 

The goal of the API is to always be fast and not block. this means that the first HTTP request for a profile would be empty but the following ones will always return data fast.

Reply:
if the API returns empty the first time, how does the consumer of this endpoint know? do they have to make a 2nd request? and when do they know when to make the 2nd request?

Reply:
They don't. If they're lucky they get some info; if not, then they only get it next time. 
Relays are extremely unreliable and one idea of this app is that it is not blocking the client. We can learn how it behaves and monitor it and then think of improvements.